.github/workflows/client.yml

```diff
@@ -20,7 +22,7 @@ jobs:
 
     - uses: actions/setup-node@v4
       with:
-        node-version: ${{ matrix.node-version }}
+        node-version: ${{ inputs.node-version }}
```

Comment:
yep I know, I didn't like that syntax and it's also a question how often it will be used with a matrix, but you can use it with one if needed:
```
jobs:
  server:
    strategy:
      matrix:
        python-version: ['3.8', '3.10', '3.12']
    uses: ./.github/workflows/server.yml
    with: 
        python-version: ${{ matrix.python-version }}
```